Update personal picture

olena-kozak opened the pull request:
![photo_2022-08-17_13-05-27](https://user-images.githubusercontent.com/98499085/185149307-48102263-ec22-4f7c-9be1-75901825b946.jpg)
When error occurred during saving picture to blog storage:
![photo_2022-08-17_13-30-22](https://user-images.githubusercontent.com/98499085/185149314-80c84736-a079-4612-b01e-b6a4cecccae1.jpg)
![photo_2022-08-17_15-26-58](https://user-images.githubusercontent.com/98499085/185149318-f0d582a8-3c91-4f59-a382-2cb5da06e6c6.jpg)
![photo_2022-08-17_16-07-37](https://user-images.githubusercontent.com/98499085/185149319-8299794d-b40a-4c3a-b616-3976c398f60b.jpg)
![photo_2022-08-17_16-22-50](https://user-images.githubusercontent.com/98499085/185149324-74d0c9e1-48c9-45db-83d2-0819658a4b6c.jpg)

When size bigger than 5MB:
![photo_2022-08-17_16-26-11](https://user-images.githubusercontent.com/98499085/185149325-e117ad17-4bb3-4905-b89c-7a8ec349fb90.jpg)

Loading images to availability page:
![photo_2022-08-17_16-35-26](https://user-images.githubusercontent.com/98499085/185149326-cf09c3b7-fb45-47bb-9d4e-a3921e8d26d4.jpg)

olena-kozak committed:
Merge branch 'development' of https://github.com/BinaryStudioAcademy/bsa-2022-easymeets into feature/update_personal_picture

# Conflicts:
#	frontend/src/app/core/services/confirmation-window.service.ts

olena-kozak committed:
Merge branch 'development' of https://github.com/BinaryStudioAcademy/bsa-2022-easymeets into feature/update_personal_picture

olena-kozak committed:
Add load image method to user-profile-page.component.ts
Add cursor: pointer style to upload-picture-button class
Add load file input to user-profile-page.component.html

olena-kozak committed:
Merge branch 'development' of https://github.com/BinaryStudioAcademy/bsa-2022-easymeets into feature/update_personal_picture

olena-kozak committed:
Merge branch 'development' of https://github.com/BinaryStudioAcademy/bsa-2022-easymeets into feature/update_personal_picture

olena-kozak committed:
Merge branch 'development' of https://github.com/BinaryStudioAcademy/bsa-2022-easymeets into feature/update_personal_picture

olena-kozak committed:
Change request type to FormData in uploading image

olena-kozak committed:
Add ability to send FormData as PUT request

olena-kozak committed:
Remove comments and unnecessary attributes

olena-kozak committed:
Remove Authorize
Add ability to receive FormData from request

olena-kozak committed:
Merge branch 'development' of https://github.com/BinaryStudioAcademy/bsa-2022-easymeets into feature/update_personal_picture

# Conflicts:
#	frontend/src/app/core/services/confirmation-window.service.ts
#	frontend/src/app/modules/user/user-profile-page/user-profile-page.component.html
#	frontend/src/app/modules/user/user-profile-page/user-profile-page.component.ts

olena-kozak committed:
Add package Microsoft.AspNetCore.Http.Features
Fix UploadFileService so that it received FormFile and save Stream to blob storage

olena-kozak committed:
Fix bug with returning imageUrl from Api

olena-kozak committed:
Add validation on picture type

olena-kozak committed:
Merge branch 'development' of https://github.com/BinaryStudioAcademy/bsa-2022-easymeets into feature/update_personal_picture

olena-kozak committed:
Add confirmation window to user-profile-page.component.ts

olena-kozak committed:
Add handle error window to user-profile-page.component.ts

olena-kozak committed:
Change button style

olena-kozak committed:
Change variable name in UserController

olena-kozak committed:
Add displaying uploaded image

olena-kozak committed:
Change user id in getCurrentUser method

olena-kozak committed:
Remove UploadImageDTO.ts

olena-kozak committed:
Remove debugger

olena-kozak committed:
Fix bug with AzureConnectionString environment variable
Remove unused methods and variables in user-profile-page.component.ts

olena-kozak committed:
Add authorize attribute to UploadImageController

olena-kozak committed:
Merge branch 'development' of https://github.com/BinaryStudioAcademy/bsa-2022-easymeets into feature/update_personal_picture

olena-kozak committed:
Add catching Exception on backend and displaying error message on frontend

olena-kozak committed:
Merge branch 'development' into feature/update_personal_picture

olena-kozak committed:
Merge branch 'development' of https://github.com/BinaryStudioAcademy/bsa-2022-easymeets into feature/update_personal_picture

tatianahutii commented on backend/EasyMeets.Core/EasyMeets.Core.WebAPI/Controllers/UploadImageController.cs:
```diff
@@ -18,16 +17,23 @@ public UploadImageController(IUploadFileService uploadFileService)
-                return NotFound();
+            var formCollection = await Request.ReadFormAsync();
+            var file = formCollection.Files.First();
+            var fileName = Path.GetFileName(file.FileName);
+            try
+            { 
+                var imageUrl = await _uploadFileService.UploadFileBlobAsync(file, fileName, 2); if (imageUrl is null)
+                {
+                    return NotFound();
+                } 
+                return Ok(new { imagePath = imageUrl });
             }
-
-            return Ok();
+            catch (Exception ex)
+            {
+                return BadRequest(ex);
+            } 
```
Why you didn't add parameters to the controller method?

olena-kozak replied:
This is FormCollection, so I decided to use Request.ReadFormAsync() to get file, so that this code is cleaner and for me looks better than using:
public async Task<IActionResult> UploadImageAsync(IFormCollection data)
        {
            string fname = data["fname_name"]; 
            ///...
        }
  And also as I understand form name should be known (["fname_name"]), I this it is better be not bind to form name 
  But I can use the second variant, if it is necessary.

tatianahutii commented on frontend/src/app/modules/user/user-profile-page/user-profile-page.component.ts:
```diff
@@ -60,9 +72,10 @@ export class UserProfilePageComponent extends BaseComponent implements OnInit {
         });
 
-        this.userService.getCurrentUserById(1)
+        this.userService.getCurrentUserById(2)
```
Please use real user

tatianahutii commented on frontend/src/app/modules/user/user-profile-page/user-profile-page.component.ts:
```diff
@@ -108,4 +123,52 @@ export class UserProfilePageComponent extends BaseComponent implements OnInit {
         this.countryCode = this.countryCodeValues[form.value.country];
     }
+
+    public loadImage({ files }: any) {
```
Please do not use any

tatianahutii commented on frontend/src/app/modules/user/user-profile-page/user-profile-page.component.ts:
```diff
@@ -108,4 +123,52 @@ export class UserProfilePageComponent extends BaseComponent implements OnInit {
+                    this.imageUrl = resp.imagePath;
+                },
+                () => {
+                    this.errorDialog();
```
It would be better to use toaster

olena-kozak committed:
Remove error dialog window, add error toast instead

olena-kozak committed:
Merge branch 'feature/update_personal_picture' of https://github.com/BinaryStudioAcademy/bsa-2022-easymeets into feature/update_personal_picture

olena-kozak committed:
Add explicit variable type to parameter in loadImage() method

olena-kozak committed:
Merge branch 'development' of https://github.com/BinaryStudioAcademy/bsa-2022-easymeets into feature/update_personal_picture

# Conflicts:
#	frontend/src/app/modules/user/user-profile-page/user-profile-page.component.html

olena-kozak committed:
Fix bug with form markup in user-profile-page.component.html

olena-kozak committed:
Remove type any to event target

olena-kozak committed:
Add parameter IFormFile file to UploadImageController
Remove hadrcoded user id from UploadController

olena-kozak committed:
Merge branch 'development' of https://github.com/BinaryStudioAcademy/bsa-2022-easymeets into feature/update_personal_picture

# Conflicts:
#	backend/EasyMeets.Core/EasyMeets.Core.WebAPI/Controllers/UserController.cs
#	frontend/src/app/modules/user/user-profile-page/user-profile-page.component.html

olena-kozak committed:
Change id type to long

olena-kozak committed:
Remove unused method

olena-kozak committed:
Merge branch 'development' into feature/update_personal_picture

olena-kozak committed:
Remove unused method form IUserService

olena-kozak committed:
Remove debugger from user-profile-page.component.ts

olena-kozak committed:
Merge branch 'development' of https://github.com/BinaryStudioAcademy/bsa-2022-easymeets into feature/update_personal_picture

# Conflicts:
#	frontend/src/app/modules/user/user-profile-page/user-profile-page.component.ts

olena-kozak committed:
Merge branch 'development' into feature/update_personal_picture

olena-kozak committed:
Merge branch 'development' of https://github.com/BinaryStudioAcademy/bsa-2022-easymeets into feature/update_personal_picture

olena-kozak committed:
Transfer GetCurrentUserEmail() method to UserService
Remove IHttpContextAccessor from UserController
Remove sending user id to beckend, add getting user from context in UploadImageAsync method

olena-kozak committed:
Merge branch 'feature/update_personal_picture' of https://github.com/BinaryStudioAcademy/bsa-2022-easymeets into feature/update_personal_picture

olena-kozak committed:
Remove async in GetCurrentUserEmail method

olena-kozak committed:
Merge branch 'development' into feature/update_personal_picture

olena-kozak committed:
Fix build

olena-kozak committed:
Merge branch 'feature/update_personal_picture' of https://github.com/BinaryStudioAcademy/bsa-2022-easymeets into feature/update_personal_picture

tatianahutii reviewed (approved):
LGTM
